src/services/utils.test.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  formatGrinAmount,
+  getFileExtension,
+  validateExtension,
+  validateUrl,
+  validateOnion,
+  validateAddress,
+} from "./utils";
+
+describe("Utils", () => {
+  test("formatGrinAmount()", () => {
+    expect(formatGrinAmount(10)).toBe(0.00000001);
+    expect(formatGrinAmount(20)).toBe(0.00000002);
+  });
+  test("getFileExtension()", () => {
+    expect(getFileExtension("0001.tx")).toBe("tx");
+    expect(getFileExtension("0001.response")).toBe("response");
+  });
+  test("validateExtension()", () => {
+    expect(validateExtension("0001.response", "response")).toBe(true);
+    expect(validateExtension("0001.response", "tx")).toBe(false);
+  });
+  test("validateUrl()", () => {
+    expect(validateUrl("http://duckduckgo.com")).toBe(true);
+    expect(validateUrl("https://duckduckgo.com")).toBe(true);
+    expect(validateUrl("http://duckduckgo.com/")).toBe(true);
+    expect(validateUrl("https://duckduckgo.com/")).toBe(true);
+    expect(validateUrl("http://duckduckgo:80")).toBe(false);
+    expect(validateUrl("http://duckduckgo.com:80")).toBe(true);
+    expect(validateUrl("https://duckduckgo.com:5000/")).toBe(true);
+  });
+  test("validateOnion()", () => {
+    expect(validateOnion("http://duckduckgo")).toBe(false);
+    expect(
+      validateOnion("jamie22ezawwi5r3o7lrgsno43jj7vq5en74czuw6wfmjzkhjjryxnid")
+    ).toBe(true);
+    expect(
+      validateOnion(
+        "http://jamie22ezawwi5r3o7lrgsno43jj7vq5en74czuw6wfmjzkhjjryxnid.onion"
```

Comment:
I'm not sure where the actual logic is, but this test just reminded me: When the user enters `http://jamie22ezawwi5r3o7lrgsno43jj7vq5en74czuw6wfmjzkhjjryxnid.onion`, we should only send `jamie22ezawwi5r3o7lrgsno43jj7vq5en74czuw6wfmjzkhjjryxnid` to the backend.

Reply:
Got it. I just added a new util: `cleanOnionURL` to clean the Onion URL. `validateOnion` validates the URL. `cleanOnionURL` will extract the address from the URL. For example:

```javascript
cleanOnionURL("http://jamie22ezawwi5r3o7lrgsno43jj7vq5en74czuw6wfmjzkhjjryxnid.onion")
```
returns `jamie22ezawwi5r3o7lrgsno43jj7vq5en74czuw6wfmjzkhjjryxnid`.